src/Core/OOConstToString.m

```diff
@@ -69,6 +69,17 @@
 	return @"UNDEFINED";
 }
 
+NSString *OOStringFromLongRangeChartMode(OOLongRangeChartMode value)
+{
+	switch (value)
+	{
+		case OOLRC_MODE_NORMAL: return @"CHART_MODE_SUNCOLOR";
+		case OOLRC_MODE_ECONOMY: return @"CHART_MODE_ECONOMY";
+		case OOLRC_MODE_GOVERNMENT: return @"CHART_MODE_GOVERNMENT";
+		case OOLRC_MODE_TECHLEVEL: return @"CHART_MODE_TECHLEVEL";
+	}
+	return @"UNDEFINED";
+}
```

Comment:
I would rather change this to something like
```
NSString *OOStringFromLongRangeChartMode(OOLongRangeChartMode value)
{
	switch (value)
	{
		#include "OOLongRangeChartMode.tbl"
	}
	
	return @"UNDEFINED";
}
```
Remember that the ENTRY macro is defined at this point, so it is convenient to use it like the neighboring functions do. See below for a possible definition of the table.